test/unit/range/iterator_test_template.hpp

```diff
@@ -685,35 +685,43 @@ inline void address_difference_test(it_begin_t it_begin, it_sentinel_t it_end)
 
 TYPED_TEST_P(iterator_fixture, address_of_end)
 {
-    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    // sentinel must not be the same as the iterator
+    if constexpr (std::ranges::common_range<decltype(this->test_range)>)
```

Comment:
Why is that?
Also, code and comment are contradictory.

Reply:
```suggestion
    // `end - begin` must be a valid expression
    using rng_t = decltype(this->test_range);
    if constexpr (std::sentinel_for<std::ranges::sentinel_t<rng_t>, std::ranges::iterator_t<rng_t>>)
```

Reply:
We talked about this, and it is a property of needing it to be common. We changed the test function declaration to reflect that fact.